### PR TITLE
Fix shuffleboard package

### PR DESCRIPTION
**Pull request by virtuald**
- Lots of places where reference_internal is needed because shuffleboard uses references extensively
- Add somewhere to hold wpi::Sendables just like SmartDashboard. Can't reuse that because they're in different packages

Blocked on https://github.com/wpilibsuite/allwpilib/pull/4884

### CODE REVIEW

File: gen/shuffleboard/ShuffleboardContainer.yml [hunk 1]
```diff
@@ -14,64 +17,148 @@ classes:
...
           std::string_view, std::span<const double>:
+            return_value_policy: reference_internal
           std::string_view, std::span<const float>:
+            return_value_policy: reference_internal
           std::string_view, std::span<const int64_t>:
+            return_value_policy: reference_internal
           std::string_view, std::span<const std::string>:
+            return_value_policy: reference_internal
       AddCamera:
         ignore: true
```
Comment (auscompgeek) [inline gen/shuffleboard/ShuffleboardContainer.yml:108]:
```suggestion
```

File: gen/shuffleboard/ShuffleboardContainer.yml [hunk 2]
```diff
@@ -14,64 +17,147 @@ classes:
...
+            param_override:
+              sendable:
+                name: defaultValue
+            cpp_code: |
+              [](ShuffleboardContainer *self, py::str &key, std::shared_ptr<wpi::Sendable> sendable) -> frc::ComplexWidget& {
+                if (!sendable) {
+                  throw FRC_MakeError(err::NullParameter, "{}", "value");
+                }
+
+                // convert key to a raw string so that we can create a StringRef
```
Comment (auscompgeek) [inline gen/shuffleboard/ShuffleboardContainer.yml:46]: Doesn't pybind11 support string_view arguments?

Comment (virtuald) [reply gen/shuffleboard/ShuffleboardContainer.yml:46]: Yes, but I need to store a py::str in the shuffeboard data anyways, so this makes it so I don't have to recreate it.

File: gen/shuffleboard/ShuffleboardContainer.yml [hunk 3]
```diff
@@ -14,64 +17,148 @@ classes:
...
           std::string_view, int:
+            return_value_policy: reference_internal
           std::string_view, std::string_view:
+            return_value_policy: reference_internal
           std::string_view, std::span<const bool>:
+            return_value_policy: reference_internal
           std::string_view, std::span<const double>:
+            return_value_policy: reference_internal
           std::string_view, std::span<const float>:
+            return_value_policy: reference_internal
```
Comment (auscompgeek) [inline gen/shuffleboard/ShuffleboardContainer.yml:158]: Will this overload be reachable? Maybe passing in a numpy float32 array could work (I haven't tried), but will it hit the `std::span<const double>` type caster before reaching this?

Comment (virtuald) [reply gen/shuffleboard/ShuffleboardContainer.yml:158]: Probably not. But really, will anyone care?

Comment (auscompgeek) [reply gen/shuffleboard/ShuffleboardContainer.yml:158]: It'll probably be confusing if we leave an unreachable thing documented.

Comment (virtuald) [reply gen/shuffleboard/ShuffleboardContainer.yml:158]: Well when someone tries to reach it they can file a bug, and then we know someone cares about it.

**Pull request merged**